### ISSUE TITLE
NextCloud, Microsoft Hyper-V and Windows Server 2016

### ISSUE THREAD

**Issue by ThorPrez** [nextcloud/vm#750]
Hi.

I´ve bought the Nextcloud for Hyper-V, because I had problems with generating of the certificate.
The funny thing was, that I had the same problems with the one from HanssonIT.
I´ve found out, that I need to forward 80 and 443 not only in the Nextcloud VM, rather on the Windows Server 2016, too.
So now both ports are forwarded for Host and Client, generating a certifivate was possible and the installation/configuration is finished.
But Nextcloud is not accessible externally, also not internally, because it´s redirected.
When I´m using the nextcloud application on Windows 10 with local IP I can up- and download files.
Port forwarding is activated by using netsh on network interface.

So, what´s the problem with Hyper-V or Windows Server 2016 ?
On ESXi it´s working fine.

With kind regards,
Thorsten

**Comment by enoch85** [comment]
Thanks for your report!

I've done several installations on Windows Server 2016 without any issues with reaching it from outside. 

Normally you forward the ports in your firewall to the specific IP of the server in question. I can't answer if you have any special setup or anything else that migtht interfere.

> I´ve found out, that I need to forward 80 and 443 not only in the Nextcloud VM,

You don't need to open any ports on the guest system, Apache will open them for you as long as the service is used, which it is if you are running the LE scripts.

> rather on the Windows Server 2016, too.

Yes, if Windows Server acts as your firewall, then you need to open them there.

> because it´s redirected.

That sounds like a misconfiguration on your end. Probably DNS, but it might be something else. The only thing the scripts will generate is a proper 80 --> 443 redirect, so that everything that hits port 80 gets redirected to port 443. That's for security.

Basically, you don't need to do anything else than open the correct ports and then run the LE (SSL) script provided.

I think this question is better suited for the forums: https://help.nextcloud.com. You could also buy support to get help.

**Comment by ThorPrez** [comment]
When I´m using **netstat -tulpen** I cann see, that Port 80 and 443 are open for TCP6, but not for TCP.
But those ports are open for Windows Server 2016.
And within the installation I got the message, that both ports were open.
So, what´s the difference between your LE script and running Nextcloud from the outside ?
And there´s no difference, if the ports are open for the vm or not.
Really necessary is to open both ports for the host.

**Comment by enoch85** [comment]
Guest system = (inside) Ubuntu 
Host = Your hypervisor
VM = Server that runs on the Hypervisor

> open or not open the ports for the vm on router

You need to open them in your router, yes. But the router is not your guest system.

> And nmap is not on your vm and not in the repository,

Just run `sudo apt install nmap`, the beauty of linux ;)

**Comment by enoch85** [comment]
> unable to locate nmap ;-)

`sudo apt update` before, to update the repositories. ;)

Again, this is a question for the forum. 

> Thanks for your report!
I've done several installations on Windows Server 2016 without any issues with reaching it from outside.

**Comment by ThorPrez** [comment]
Nmap scan report for localhost (127.0.0.1)
Host is up (0.000017s latency).
Other addresses for localhost (not scanned): ::1
Not shown: 995 closed ports
PORT      STATE SERVICE
22/tcp    open  ssh
80/tcp    open  http
443/tcp   open  https
5432/tcp  open  postgresql
10000/tcp open  snet-sensor-mgmt

Nmap done: 1 IP address (1 host up) scanned in 1.70 seconds

**Comment by enoch85** [comment]
> 80/tcp open http
443/tcp open https

There you go, no need to open it in the guest as you proposed. :)

> I´ve found out, that I need to forward 80 and 443 not only in the Nextcloud VM

**Comment by ThorPrez** [comment]
In my opinion it´s a question for the developer, because I´ve bought a product and it´s not working. As you can see both ports are open and I can´t reach my cloud. If there´re some adjustments on the server necessary, it would be helpful to know it.
There´s no ducumentation for this variant of using nextcloud with Hyper-V, except those not-working tutorials in the web. That´s the reason why I´ve bought your product.
It´s also not working in Virtualbox !

**Comment by enoch85** [comment]
I think it's strange that it works for me but not you.

I could spend 1-2 hours just to prove my cause by making a video installing it, but right now I have to much work, and I don't have time. Sorry.

If this was a bug I would attend with 100% concentration and engagement, but as I know for a fact that it works for me it seems more like a user misconfiguration, which is why I recommended the forum, or to buy support.

> As you can see both ports are open and I can´t reach my cloud

What you showed me was that they are open in in the guest system, which is always the case when there are Apache Vhosts configured, you'd still need to open ports in your firewall router to be able to reach it from the outside, as I mentioned before. [This guide](https://docs.hanssonit.se/s/W6fMouPiqQz3_Mog/virtual-machines-vm/d/W6-83ePiqQz3_MrT/publish-your-server-online) is kind of outdated but explains the basics at least. 

I hope it helps!
